src/resources/startup.py

```diff
@@ -293,10 +297,45 @@ def Execute(self, request, context):
         return execute_response
 
 
+def watch_connections(address, event):
+    number_of_failures = 0
+    timeout_sec = 1
+
+    while True:
```

Comment:
So this makes sure that a connection is established within a timeout after the server starts? Is there a particular reason it is done this way? Why wait for 3 failures? If I have success, failure, failure, then things are ok? That's doesn't seem right. If the timeout is 12 seconds, why not just wait 12 seconds and then check once?

Reply:
The goal here is to check that always there is a socket other than this socket connect to the gRPC server. Two things can happen when there is a problem. First, the Python GRPC server may have stopped. Or the Triton might have exited non-gracefully. If the first case happened, the connection to the Python GRPC will fail. I just added this to make sure that there is not a temporary problem with the connection. I can probably reduce the number of failures to 2. But in the example you mentioned, if you have success, failure, failure, failure, then the Python backend will exit.

Reply:
I see... what you have is ok then.